tools/build.py

```
@@ -211,15 +211,16 @@
     successes = []
     skipped = []
 
-    # CPPCHECK code validation
-    if options.cppcheck_validation:
-        for toolchain in toolchains:
-            if not TOOLCHAIN_CLASSES[toolchain].check_executable():
-                search_path = TOOLCHAIN_PATHS[toolchain] or "No path set"
-                args_error(parser, "Could not find executable for %s.\n"
-                                   "Currently set search path: %s"
-                           % (toolchain, search_path))
-            for target in targets:
+    for toolchain in toolchains:
+        if not TOOLCHAIN_CLASSES[toolchain].check_executable():
```

Could we do these checks all up front? to fail fast.